students/larry_mburu/Lesson04/mailroom.py

```diff
@@ -0,0 +1,233 @@
+    """
+    Function, sends a thank you letter to all donor's in the donor DB list
+
+    param: validation: If set to True, prints the contents of single donor file, 
+    since tempfile.TempDirectory() does not persist files, in the file system. 
+    It's purely to validate that the letter / formatting was written to file
+    correctly.
+
+    return: None 
+    """
+
+    with tempfile.TemporaryDirectory() as file_tmp_dir: 
+        for donor in donations_per_individual:  
+
+            # -1, the latest donation in the donor "list"
+            email_template = f"""
+            Dear {donor}, 
+                Thank you for your kind donation of ${donations_per_individual[donor][-1]} 
+
+                it will be put to very good use. 
+
+                Sincerely,
+                    - The Cloud Squad 
+            """
+            with open(file_tmp_dir + os.sep + donor + '.txt', 'w') as fobj: 
+                fobj.write(email_template)
+
+        if validate == True: 
+            letter_validator(file_tmp_dir)
+
+def send_thank_you_email(name, amount): 
+    """
+    composes an email to a selected donor on 
+    the donor's list 
+    """
+    print(f"Thank you {name} for your generous donation of $ {amount:.2f} dollars")
+    print()
+
+    #return to the original prompt 
+    main()
+
+def prompt_for_donor_name(): 
+    """
+    Function, asks user for a donar name. The user can opt to
+    list all the donors in the donor DB list. 
+    """
+
+    donor_name = input("Enter donor's full name or 'list' to list donors: ") 
+    quit = True
+    while quit and donor_name == 'list':
+        for donar in donations_per_individual: 
+            print(donar[0]) 
+        donor_name = input("Enter donar full name or 'list' to list donars: ")
+        quit = False
+    return donor_name
+
+def add_donar_to_list():
+    """
+    After user is prompted for donor name, function checks if 
+    it's a known or unknown donor. If unknown, a new record is 
+    added to the main donation DB list
+
+    return: if donor is in the DB list or not, function returns the donor name 
+    for further use
+    """
+
+    donor = prompt_for_donor_name()
+    donor_list = []
+
+    #extract donor names from main donation DB list for in checks. 
+    for key in donations_per_individual: 
+        donor_list.append(key)
+    
+    # add unknown donor to the main donation DB list
+    if donor not in donor_list:
+        donations_per_individual[donor] = []
+    
+    return donor
+
+def prompt_for_donation_amount():
+    """
+    function asks user for donor, donar amount, 
+    and proceeds to send a thank you to the donor
+    """
+
+    donor_name = add_donar_to_list()
+    response = input(f"How much would {donor_name} like to donate? ")
+    donation_amount = float(response) 
+
+    #add donation to donor history, in donation DB list
+    if donor_name in donations_per_individual: 
+        donations_per_individual[donor_name].append(donation_amount)
```

Comment:
same comment here, no need to loop - just find donor item by accessing the key in the dictionary.